executor/stale_txn_test.go

```diff
@@ -235,47 +236,89 @@ func (s *testStaleTxnSerialSuite) TestSelectAsOf(c *C) {
 
 func (s *testStaleTxnSerialSuite) TestStaleReadKVRequest(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
+	safePointName := "tikv_gc_safe_point"
+	safePointValue := "20160102-15:04:05 -0700"
+	safePointComment := "All versions after safe point can be accessed. (DO NOT EDIT)"
+	updateSafePoint := fmt.Sprintf(`INSERT INTO mysql.tidb VALUES ('%[1]s', '%[2]s', '%[3]s')
+	ON DUPLICATE KEY
+	UPDATE variable_value = '%[2]s', comment = '%[3]s'`, safePointName, safePointValue, safePointComment)
+	tk.MustExec(updateSafePoint)
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists t")
+	tk.MustExec(`drop table if exists t1`)
+	tk.MustExec(`drop table if exists t2`)
 	tk.MustExec("create table t (id int primary key);")
+	tk.MustExec(`create table t1 (c int primary key, d int,e int,index idx_d(d),index idx_e(e))`)
 	defer tk.MustExec(`drop table if exists t`)
+	defer tk.MustExec(`drop table if exists t1`)
+	conf := *config.GetGlobalConfig()
+	oldConf := conf
+	defer config.StoreGlobalConfig(&oldConf)
+	conf.Labels = map[string]string{
+		placement.DCLabelKey: "sh",
+	}
+	config.StoreGlobalConfig(&conf)
 	testcases := []struct {
-		name     string
-		sql      string
-		txnScope string
-		zone     string
+		name string
+		sql  string
 	}{
 		{
-			name:     "coprocessor read",
-			sql:      "select * from t",
-			txnScope: "local",
-			zone:     "sh",
+			name: "coprocessor read",
+			sql:  "select * from t",
 		},
 		{
-			name:     "point get read",
-			sql:      "select * from t where id = 1",
-			txnScope: "local",
-			zone:     "bj",
+			name: "point get read",
+			sql:  "select * from t where id = 1",
 		},
 		{
-			name:     "batch point get read",
-			sql:      "select * from t where id in (1,2,3)",
-			txnScope: "local",
-			zone:     "hz",
+			name: "batch point get read",
+			sql:  "select * from t where id in (1,2,3)",
 		},
 	}
+	failpoint.Enable("tikvclient/assertStoreLabels", fmt.Sprintf(`return("%v_%v")`, placement.DCLabelKey, "sh"))
+	failpoint.Enable("tikvclient/assertStaleReadFlag", `return(true)`)
 	for _, testcase := range testcases {
 		c.Log(testcase.name)
-		tk.MustExec(fmt.Sprintf("set @@txn_scope=%v", testcase.txnScope))
-		failpoint.Enable("github.com/pingcap/tidb/config/injectTxnScope", fmt.Sprintf(`return("%v")`, testcase.zone))
```

Comment:
Can this failpoint be removed?

Reply:
This failpoint is unnecessary now, we directly updated config now.

Reply:
Yeah, I mean, can you remove the related failpoint injection code?

Reply:
Found that `github.com/pingcap/tidb/config/injectTxnScope` is renamed to `tikvclient/injectTxnScope`. And there is actually two `failpoint.Enable("github.com/pingcap/tidb/config/injectTxnScope"`. Another one is in `TestExactStalenessTranasction`.

Reply:
I don't think so, it is also useful in other test cases.

Reply:
But there is no `failpoint.Inject("github.com/pingcap/tidb/config/injectTxnScope")`. So it is actually a stab.

Reply:
updated.